File: C/maximum_depth_of_binary_tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

#include <math.h>

int maxDepth(struct TreeNode* root){
    if(root == NULL) return 0;
    else {
        return 1 + (int)fmax(maxDepth(root->left), maxDepth(root->right));
    }
}
        
int countDepth(struct TreeNode* p, struct TreeNode* q) {
    if(p == NULL && q == NULL) return 0;
    else if(p != NULL && q == NULL) {
        return 1 + countDepth(p->left, p->right);
    }
    else if(p == NULL && q != NULL) {
        return 1 + countDepth(q->left, q->right);
    }
    else {
        return 1 + (int)fmax(countDepth(p->left, p->right), countDepth(q->left, q->right));
    }
}

File: Python3/maximum_depth_of_binary_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:    
        if not root: return 0
        else: 
            depth = 1
            return self.countDepth(root.left, root.right, depth)
        
    def countDepth(self, p, q, num):
        if not p and not q: return num
        elif p and not q:
            num += 1
            return self.countDepth(p.left, p.right, num)
        elif not p and q:
            num += 1
            return self.countDepth(q.left, q.right, num)
        else:
            num += 1
            return max(self.countDepth(p.left, p.right, num), self.countDepth(q.left, q.right, num))
